katsu_ingest.py

```diff
@@ -263,37 +282,38 @@ def ingest_donor_with_clinical(katsu_server_url, dataset, headers):
         response = requests.Session().send(request.prepare())
         if response.status_code != HTTPStatus.CREATED:
             if "unique" in response.text:
-                return IngestCohortException(
-                    f"Program {program_id} has already been ingested into Katsu. "
-                    "Please delete and try again."
+                errors.append(
+                    f"Program {program_id} has already been ingested into Katsu. Please delete and try again."
                 )
             else:
-                return IngestServerException(
+                errors.append(
                     [
                         f"\nREQUEST STATUS CODE: {response.status_code}"
                         f"\nRETURN MESSAGE: {response.text}\n"
                     ]
                 )
-    else:
-        return IngestCohortException(f"No ingestable donors found in dataset.")
-    for donor in dataset["donors"]:
-        parents = [("programs", program_id)]
-        print(f"Loading donor {donor['submitter_donor_id']}...")
-        try:
-            ingested_ids = {}
-            traverse_clinical_field(fields, donor, "donors", parents, types, ingested_ids)
-            # print(json.dumps(ingested_ids, indent=2))
-        except Exception as e:
-            print(traceback.format_exc())
-            return IngestServerException(str(e))
-    fields.pop("programs")
-    print(json.dumps(fields, indent=4))
-    errors = ingest_fields(fields, katsu_server_url, headers)
-    if errors:
-        return IngestServerException(errors)
-    else:
-        return IngestResult(len(dataset["donors"]))
+            continue
+        fields = {type: [] for type in types}
+
+        for donor in donors:
+            parents = [("programs", program_id)]
+            print(f"Loading donor {donor['submitter_donor_id']}...")
+            try:
+                ingested_ids = {}
+                traverse_clinical_field(fields, donor, "donors", parents, types, ingested_ids)
+                # print(json.dumps(ingested_ids, indent=2))
+            except Exception as e:
+                print(traceback.format_exc())
+                errors.append(str(e))
+        fields.pop("programs")
+        print(json.dumps(fields, indent=4))
```

Comment:
The printout can get too long in the console, maybe save for debug only?

Reply:
I think that it's fine for now; it's more useful than not for debugging ingests.